lnrpc/routerrpc/router_backend.go

```diff
@@ -522,9 +522,14 @@ func (r *RouterBackend) extractIntentFromSendRequest(
 
 	payIntent := &routing.LightningPayment{}
 
-	// Pass along an outgoing channel restriction if specified.
+	// Pass along restrictions on the outgoing channels that may be used.
+	payIntent.OutgoingChannelIDs = rpcPayReq.OutgoingChanIds
+
+	// Add the deprecated single outgoing channel restriction if present.
 	if rpcPayReq.OutgoingChanId != 0 {
-		payIntent.OutgoingChannelID = &rpcPayReq.OutgoingChanId
+		payIntent.OutgoingChannelIDs = append(
```

Comment:
Could also fail if both are set? Take the deprecation of the old field seriously if the caller is using the new one?

Reply:
Yes, I want to be serious about it. Fixed